### PR TITLE
[40] Calculate Product Shipping Cost 

### PR DESCRIPTION
**Pull request by Samju200**
## Task: `calculation for shipping cost and total product cost`
- `Changes you made to given task`

### CODE REVIEW

File: submissions/adeyemiJuwon/dom/product.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+let quantity = document.getElementById("quantity");
+// let price = document.getElementById("price");
+let shippingCost = document.getElementById("shippingCost");
+let totalCost = document.getElementById("totalCost");
+let button = document.getElementById("btn");
+
+
+let itemPrice = 900;
```
Comment (ToluAkin) [inline submissions/adeyemiJuwon/dom/product.js:8]: You can set item price here. You have to get the value from the html  just like `quantity`  @Samju200

File: submissions/adeyemiJuwon/dom/product.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		shipping = 25000
+		shippingCost.innerHTML = "£" + shipping + " GBP";
+	}else{
+		shipping = 100
+		shippingCost.innerHTML = "£" + shipping + " GBP";
+	}
+	return true
+}
+
+function buttonDeactived(){
```
Comment (ToluAkin) [inline submissions/adeyemiJuwon/dom/product.js:47]: You are suppose to disable the button not change the color @Samju200. The intention is the user would not be able to click by now

**Review by ToluAkin — CHANGES_REQUESTED** [review]
@Samju200 There are other comments from the automated reviewer that you should attend to as well.

**Comment by ToluAkin** [comment]
@Samju200 The success criteria for this task has been updated. Please check again https://github.com/Dufuna-CodeCamp/Dufuna-CodeCamp20/issues/479

File: submissions/adeyemiJuwon/dom/product.js (hunk 1 above)
Comment (Samju200) [reply submissions/adeyemiJuwon/dom/product.js]: Yes
Thank You 
Done @ToluAkin

File: submissions/adeyemiJuwon/dom/product.js (hunk 2 above)
Comment (Samju200) [reply submissions/adeyemiJuwon/dom/product.js:45]: Okay
Done

File: submissions/adeyemiJuwon/dom/product.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+
+
+function shipp(){
+	grossPrice = quantity.value * Number(price.innerHTML);
+	if ( grossPrice < 1000){
+		shipping = 100
+		shippingCost.innerHTML = "£" + shipping + " GBP";
+	}
+	else if(grossPrice >= 1000 && grossPrice < 20000){
```
Comment (ToluAkin) [inline submissions/adeyemiJuwon/dom/product.js:17]: `grossPrice <= 20000` @Samju200

File: submissions/adeyemiJuwon/dom/product.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+		shipping= 0.15 * grossPrice
+		shippingCost.innerHTML = "£" + shipping + " GBP";
+	}else if (grossPrice >= 20000 && grossPrice < 100000){
```
Comment (ToluAkin) [inline submissions/adeyemiJuwon/dom/product.js:20]: grossPrice <= 100000 @Samju200

File: submissions/adeyemiJuwon/dom/product.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+	}
+	return true
+	
+	
+}
+
+
+function lessThanOneInteger(){
+	if(quantity.value < 1){
+		quantity.value = 1;
```
Comment (ToluAkin) [inline submissions/adeyemiJuwon/dom/product.js:48]: Remove this line. The user needs to change the value by them self.

File: submissions/adeyemiJuwon/dom/product.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+		button.disabled = true;
+	}else{
+		button.disabled = false;
+		quantity.nextElementSibling.innerHTML = ""
+	}
+
+}
+
+function shippingTotalCost(e){
+	lessThanOneInteger(e)
```
Comment (ToluAkin) [inline submissions/adeyemiJuwon/dom/product.js:59]: You are not using `event ` in any of the functions @Samju200

**Review by ToluAkin — CHANGES_REQUESTED** [review]

File: submissions/adeyemiJuwon/dom/product.js (hunk 3 above)
Comment (ToluAkin) [reply submissions/adeyemiJuwon/dom/product.js:17]: @Samju200

**Review by ToluAkin — CHANGES_REQUESTED** [review]

File: submissions/adeyemiJuwon/dom/product.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+function shippingTotalCost(){
+	lessThanOneInteger();
+	shipp();
+	cost();
+}
+
+document.addEventListener('DOMContentLoaded', function(){
+    
+	customer.addEventListener('submit', function(e){
```
Comment (ToluAkin) [inline submissions/adeyemiJuwon/dom/product.js:65]: There is a missing function here @Samju200

**Review by ToluAkin — CHANGES_REQUESTED** [review]

File: submissions/adeyemiJuwon/dom/product.js (hunk 7 above)
Comment (Samju200) [reply submissions/adeyemiJuwon/dom/product.js:65]: @ToluAkin Done

File: submissions/adeyemiJuwon/dom/product.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+		shipping = 25000
+		shippingCost.innerHTML = "£" + shipping + " GBP";
+	}
+	return true
+}
+
+
+
+function cost(){
+
```
Comment (ToluAkin) [inline submissions/adeyemiJuwon/dom/product.js:33]: shipping is not defined in this function @Samju200

**Review by ToluAkin — CHANGES_REQUESTED** [review]

File: submissions/adeyemiJuwon/dom/product.js [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+function shippingTotalCost(){
+	lessThanOneInteger();
+	shipp();
+	cost();
+}
+
+document.addEventListener('DOMContentLoaded', function(){
```
Comment (ToluAkin) [inline submissions/adeyemiJuwon/dom/product.js:63]: This listener will only run when the page loads nothing else after. `input` will be better for it. @Samju200

**Review by ToluAkin — CHANGES_REQUESTED** [review]

File: submissions/adeyemiJuwon/dom/product.js (hunk 8 above)
Comment (Samju200) [reply submissions/adeyemiJuwon/dom/product.js:34]: Okay.
Done

File: submissions/adeyemiJuwon/dom/product.js (hunk 9 above)
Comment (Samju200) [reply submissions/adeyemiJuwon/dom/product.js]: Done @ToluAkin 
Thank you

File: submissions/adeyemiJuwon/dom/product.js [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+		shipping = 25000
+		shippingCost.innerHTML = "£" + shipping + " GBP";
+	}
+	return true
+}
+
+
+
+function cost(){
+	shipp();
```
Comment (ToluAkin) [inline submissions/adeyemiJuwon/dom/product.js:33]: Calling the ship function does not make shipping variable and its value available. You might want to consider joining the 2 functions. Also, when you make changes you want to test that your code does what it suppose to do. @Samju200

**Review by ToluAkin — CHANGES_REQUESTED** [review]

File: submissions/adeyemiJuwon/dom/product.js (hunk 10 above)
Comment (Samju200) [reply submissions/adeyemiJuwon/dom/product.js:33]: Yes, but the changes still work perfectly with what i want . declaring shipping variable in the function inside the cost function will not work expect i join the two function like you suggest. but i thought calling the function inside the another make every information declare in the  the function available to the function.

**Review by ToluAkin — APPROVED** [review]

**Pull request merged**